# **Домашняя работа к уроку 6. Сравнение долей. Построение доверительного интервала**

---

**1)Известно, что генеральная совокупность распределена нормально, со средним квадратическим отклонением, $𝜎 = 16$. Найти доверительный интервал для оценки математического ожидания с надежностью $0.95$, если выборочная средняя $M=80$, а объем выборки $n=256$**



Поскольку нам известно среднее квадратическое отклонение генеральной совокупности $σ = 16$, то то для расчета $95%$-го доверительного интервала, будем использовать $Z$-критерий. И границы доверительного интервала расчитаем по формуле: $\left( \bar{X} - z_{\alpha/2} \cdot \frac{\sigma}{\sqrt{n}}, \bar{X} + z_{\alpha/2} \cdot \frac{\sigma}{\sqrt{n}} \right)$, где:

$\bar{X} = M = 80$ -среднее значение выборки

$z_{\alpha/2}$ табличное значение $Z$-критерия, и для $95$%-го доверительного интервала $z_{\alpha/2} = 1.96$

$σ = 16$ - среднее квадратическое отклонение генеральной совокупности

$n = 256$ - объем выборки

Таким образом $95$%-й доверительный интервал расчитаем как:



In [2]:
left=80-1.96*16/256**(1/2)
right=80+1.96*16/256**(1/2)
print(f'95%-й доверительный интервал для оценки мат. ожидания генеральной совокупности: [{left};{right}].')

95%-й доверительный интервал для оценки мат. ожидания генеральной совокупности: [78.04;81.96].


**Ответ:**

$[78.04; 81.96]$

**2)В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95**

Поскольку из условия задачи нам ничего не известно ни о мат. ожидании генеральной совокупности ни о среднем квадратическом отклонении для генеральной совокупности, то для расчета $95$%-го доверительного интервала будем использовать $t$-критерий, и формулу: $\left( \bar{X} - t_{\alpha/2} \cdot \frac{\sigma_{несмещ}}{\sqrt{n}}, \bar{X} + t_{\alpha/2} \cdot \frac{\sigma_{несмещ}}{\sqrt{n}} \right)$, где:

$\sigma_{несмещ}$ - несмещённое среднее квадратическое отклонение выборки

$n$ - размер выборки

In [3]:
import numpy as np
arr=np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
print(f'Среднее выборочное: {np.mean(arr): .2f},\n'
      f'Размер выборки n={len(arr)},\n'
      f'Среднее квадратическое отклонение по выборке(несмещенное): {np.std(arr, ddof=1): .2f}.'
     )

Среднее выборочное:  6.59,
Размер выборки n=10,
Среднее квадратическое отклонение по выборке(несмещенное):  0.45.


In [4]:
import scipy.stats as stats
def t_from_table(confidens, len_array):
    alpha=(1-confidens)
    return stats.t.ppf(1-alpha/2, len_array-1)
print(f'Табличное значение t-критерия для 95%-го доверительного интервала данной выборки: {t_from_table(0.95, len(arr)): .3f}')

Табличное значение t-критерия для 95%-го доверительного интервала данной выборки:  2.262


Итоговая функция для расчета доверительного интервала:

In [5]:
def confidens_int(arr, confidens):
    return round(np.mean(arr)-t_from_table(confidens,len(arr))*np.std(arr, ddof=1)/len(arr)**0.5,3), \
           round(np.mean(arr)+t_from_table(confidens,len(arr))*np.std(arr, ddof=1)/len(arr)**0.5,3)

print(f'95%-й доверительный интервал для истинного значения Х: {confidens_int(arr, 0.95)}.')

95%-й доверительный интервал для истинного значения Х: (6.268, 6.912).


**Ответ:**

$[6.268, 6.912]$

**3)Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170 Рост матерей  178, 165, 165, 173, 168, 155, 160, 164, 178, 175 Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.**

In [6]:
import numpy as np
from scipy import stats

# Данные: рост матерей и их дочерей
mothers_height = [178, 165, 165, 173, 168, 155, 160, 164, 178, 175]
daughters_height = [175, 167, 154, 174, 178, 148, 160, 167, 169, 170]

# Разность ростов
differences = np.array(mothers_height) - np.array(daughters_height)

# Среднее значение разности
mean_diff = np.mean(differences)

# Стандартная ошибка среднего разности
std_err = stats.sem(differences)

# 95% доверительный интервал
confidence_interval = stats.t.interval(0.95, len(differences)-1, loc=mean_diff, scale=std_err)

print(f"Средняя разность: {mean_diff}")
print(f"95% доверительный интервал: {confidence_interval}")


Средняя разность: 1.9
95% доверительный интервал: (-2.6425007065225032, 6.4425007065225035)


**Ответ:**

$[-2.64, 6.44]$